specs/flattener.process.spec.js

```diff
@@ -0,0 +1,88 @@
+var flattener = require("../source/flattener.process.js");
+
+describe("flattener.process.js", function()
+{
+    it("\n" +
+        "AAAA\n" +
+        "----\n" +
+        "AAAA", function()
+    {
+        var input =  [createTestClip("AAAA", 0)];
+        var output = [createTestClip("AAAA", 0)];
+
+        expect(flattener.flatten(input)).toEqual(output);
+    });
+
+    it ("\n" + 
+        "AAAABBBB\n" +
+        "--------\n" +
+        "AAAABBBB", function()
+    {
+        var input =  [createTestClip("AAAA", 0), createTestClip("    BBBB", 0)];
+        var output = [createTestClip("AAAA", 0), createTestClip("    BBBB", 0)];
+
+        expect(flattener.flatten(input)).toEqual(output);
+    });
+
+    it ("\n" +
+        "AAAA    \n" +
+        "    BBBB\n" +
+        "--------\n" +
+        "AAAABBBB", function()
+    {
+        var input =  [createTestClip("AAAA", 0), createTestClip("    BBBB", 1)];
+        var output = [createTestClip("AAAA", 0), createTestClip("    BBBB", 0)];
+
+        expect(flattener.flatten(input)).toEqual(output);
+    });
+
+    it ("\n" +
+        "AAAA    BBBB\n" +
+        "------------\n" +
+        "AAAA0000BBBB", function()
+    {
+        var input =
+        [
+            createTestClip("AAAA", 0),
+            createTestClip("        BBBB", 0)
+        ];
+        var output = 
+        [
+            createTestClip("AAAA", 0),
+            createTestClip("    0000", 0),
+            createTestClip("        BBBB", 0)
+        ];
+
+        expect(flattener.flatten(input)).toEqual(output);
+    });
+});
+
+
+// Creates a test clip by looking at code string and track integer
+function createTestClip(code, track)
```

Comment:
Ba ha ha! I love this function. It takes the derpy ascii stuff we wrote in the GoogleDoc, and translates it into json.